app/models/user.rb

```diff
@@ -1,6 +1,19 @@
 class User < ActiveRecord::Base
+  has_one :beard
+
+  validates :first_name, presence: true
+  validates :last_name,  presence: true
+
+  def self.create_from_params(params)
+    user = self.new
+    user.first_name = params[:first_name]
+    user.last_name  = params[:last_name]
+    user.save
+
+    user
+  end
 
   def fullname
-    first_name + " " + last_name
+    first_name.to_s + " " + last_name.to_s
```

Comment:
Aren't you confident that the names would be string? Why not do interpolation?

```ruby
  "#{first_name} #{last_name}"
```

Or join that converts objects to strings before joining them.

```ruby
  [first_name, last_name].join(" ")
```